### PR TITLE
DOC: Roadmap documentation

### PR DESCRIPTION
**Pull request by arokem**
The idea behind this PR is to discuss a road-map and document it.

### CODE REVIEW

**Comment by arokem** [comment]
OK - this ready for review

File: README.rst [hunk 1]
```diff
@@ -36,15 +36,13 @@ of results.
 
 The workflow is based on `Nipype <https://nipype.readthedocs.io>`__ and encompases a large
 set of tools from well-known neuroimaging packages, including
-`FSL <https://fsl.fmrib.ox.ac.uk/fsl/fslwiki/>`__,
 `ANTs <https://stnava.github.io/ANTs/>`__,
 `FreeSurfer <https://surfer.nmr.mgh.harvard.edu/>`__,
-`AFNI <https://afni.nimh.nih.gov/>`__,
-and `Nilearn <https://nilearn.github.io/>`__.
+and `DIPY <https://dipy.org/>`__.
```
Comment (oesteban) [inline README.rst]: Should we just not mention any of these here (so long the citations are clearly stated in the boilerplate)?

File: README.rst [hunk 2]
```diff
@@ -36,15 +36,13 @@ of results.
...
 This pipeline was designed to provide the best software implementation for each state of
 preprocessing, and will be updated as newer and better neuroimaging software becomes
 available.
 
-dMRIPrep performs basic preprocessing steps (coregistration, normalization, unwarping,
-segmentation, skullstripping etc.) providing outputs that can be
-easily submitted to a variety of tractography algorithms.
+dMRIPrep performs basic preprocessing steps (head motion correction,
```
Comment (oesteban) [inline README.rst]:
```suggestion
dMRIPrep performs basic preprocessing steps (:abbr:`HMC (Head-Motion Correction)`,
```

File: README.rst [hunk 3]
```diff
@@ -36,15 +36,13 @@ of results.
...
+susceptibility distortion correction, eddy current correction,  etc.) providing
```
Comment (oesteban) [inline README.rst]:
```suggestion
:abbr:`SDC (Susceptibility-derived Distortion Correction)`,
:abbr:`ECC (Eddy-Currents Correction)`,  etc.) providing
```

File: docs/roadmap.rst [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+Version 0.3 (Targetted for March 1st, 2020)
+-------------------------------------------
+This version should be considered an early alpha of the software, but will
+contain a full pipeline of processing from a raw BIDS dataset to analyzable data.
+
+At this point, the processing pipeline will include the following major steps:
+
+#. Susceptibility distortion correction.
+    Using SDC flows
```
Comment (oesteban) [inline docs/roadmap.rst]:
```suggestion
    Using `SDCFlows <https://github.com/poldracklab/sdcflows>`__
```

File: docs/roadmap.rst [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+Version 0.4 (April 1st, 2020)
+-----------------------------
+#. Head motion correction.
+
```
Comment (oesteban) [inline docs/roadmap.rst]:
```suggestion

    A SHOREline-based approach, ported from QSIPREP.

```

File: docs/roadmap.rst [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+#. Eddy current correction.
+
+    For this release, these two will be done doing something related to the
```
Comment (oesteban) [inline docs/roadmap.rst]:
```suggestion
    We will explore the possible adaptations of the HMC based on SHOREline above.
```

File: docs/roadmap.rst [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    SHOREline approach In cases where the data are "shelled", 3dSHORE will be
```
Comment (oesteban) [inline docs/roadmap.rst]:
```suggestion
    In cases where the data are "shelled", 3dSHORE will be
```

Comment (oesteban) [reply docs/roadmap.rst]: Perhaps move these three lines up, under HMC?

File: docs/roadmap.rst [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+    used as the diffusion model. If the data are single-shell, we will use SFM
+    as the diffusion model.
+
+#. Identification of outliers.
```
Comment (oesteban) [inline docs/roadmap.rst]:
```suggestion
#. Framewise-displacement calculation
```

I think the definition of outlier is too vague for a roadmap item.

File: docs/roadmap.rst [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+
+Version 0.5 (May 1st, 2020)
+----------------------------
+
+#. Registration between dMRI and T1w image.
+
+#. Identification of outlier measurements (+ imputation?)
```
Comment (oesteban) [inline docs/roadmap.rst]: is this outlier in the sense of head motion? we should be more clear about that

File: docs/roadmap.rst [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+
+- Gibbs ringing (using DIPY's image-based implementation).
+
+- Denoising (e.g., MP-PCA)
+
+- Rician bias correction
+
+- Gradient non-linearity correction
+
+- Bias field correction
```
Comment (oesteban) [inline docs/roadmap.rst]:
```suggestion
- B1 inhomogeneity field estimation and :abbr:`INU (intensity non-uniformity) correction`
```

File: docs/roadmap.rst [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+Version 0.3 (Targetted for March 1st, 2020)
+-------------------------------------------
+This version should be considered an early alpha of the software, but will
+contain a full pipeline of processing from a raw BIDS dataset to analyzable data.
+
+At this point, the processing pipeline will include the following major steps:
+
+#. Susceptibility distortion correction.
+    Using SDC flows
+
```
Comment (oesteban) [inline docs/roadmap.rst]: I would add here:
```
#. Signal drift estimation
    Leveraging the :math:`b=0` extraction, rescaling and averaging that was merged in `#50 <https://github.com/nipreps/dmriprep/pull/50>`__
```

File: README.rst [hunk 12]
```diff
@@ -36,15 +36,13 @@ of results.
...
+outputs that can be easily submitted to a variety of tractography algorithms.
```
Comment (oesteban) [inline README.rst]:
```suggestion
outputs that can be easily submitted to a variety of diffusion models.
```

File: docs/roadmap.rst [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+------------------------------------------
+
+After integrating the above steps, we will spend the time leading to a 1.0
+testing the software on various datasets, evaluating and validating the
+resulting derivatives.
+
+
+Long-term plans
+---------------
+
```
Comment (oesteban) [inline docs/roadmap.rst]: please remove line

File: docs/roadmap.rst [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+#. Identification of outlier measurements (+ imputation?)
+
+If we get around to doing thesee steps earlier, they can also be included in
+earlier releases.
+
+
+Version 1.0 (Targetted for September 2020)
+------------------------------------------
+
```
Comment (oesteban) [inline docs/roadmap.rst]: please remove line

File: docs/roadmap.rst [hunk 15]
```diff
@@ -0,0 +1,72 @@
...
+resulting derivatives.
+
+
+Long-term plans
+---------------
+
+In the long run we would like to explore the following processing steps:
+
+- Gibbs ringing (using DIPY's image-based implementation).
+
```
Comment (oesteban) [inline docs/roadmap.rst]: Please remove newlines between enumeration items.

File: docs/roadmap.rst [hunk 16]
```diff
@@ -0,0 +1,72 @@
+.. include:: links.rst
+
+--------------------
+Development road map
+--------------------
+
```
Comment (oesteban) [inline docs/roadmap.rst]: I believe this newline should also go away

File: README.rst (hunk 1 above)
Comment (arokem) [reply README.rst]: Yes. I think that would be fine. Implemented in 5431efc

File: docs/roadmap.rst (hunk 9 above)
Comment (arokem) [reply docs/roadmap.rst]: That, but also other artifacts (e.g., dropped slices, zippers and so on). I've tried to clarify in 41ee304

**Comment by arokem** [comment]
Thanks for the review @oesteban! I believe I've addressed your comments and also rebased on master. Apologies for all the merge commits. I am still not used to working with the "accept suggestion" feature.

**Comment by arokem** [comment]
Hmm. I think that we might be running into issues including  certain sphinx directives in the README file, because it's also being used as the `long_description`: https://app.circleci.com/jobs/github/nipreps/dmriprep/650. Should I remove these abbreviations here?

**Comment by oesteban** [comment]
> Should I remove these abbreviations here?

Sure!

**Pull request merged**